lib/Lower/CFGBuilder.h

```diff
@@ -1,358 +0,0 @@
-//===-- CFGBuilder.h -- CFG building parts of bridge ------------*- C++ -*-===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-#ifndef FORTRAN_LOWER_CFGBUILDER_H_
-#define FORTRAN_LOWER_CFGBUILDER_H_
-
-/// Traverse the PFT and complete the CFG by drawing the arcs, pruning unused
-/// potential targets, making implied jumps explicit, etc.
-class CfgBuilder {
-
-  Fortran::lower::pft::Evaluation *
-  getEvalByLabel(const Fortran::parser::Label &label) {
-    auto iter = labels.find(label);
-    if (iter != labels.end()) {
-      return iter->second;
-    }
-    return nullptr;
-  }
-
-  /// Collect all the potential targets and initialize them to unreferenced
-  void
-  resetPotentialTargets(std::list<Fortran::lower::pft::Evaluation> &evals) {
-    for (auto &e : evals) {
-      if (e.isTarget) {
-        e.isTarget = false;
-      }
-      if (e.lab.has_value()) {
-        labels.try_emplace(*e.lab, &e);
-      }
-      if (e.subs) {
-        resetPotentialTargets(*e.subs);
-      }
-    }
-  }
-
-  /// cache ASSIGN statements that may yield a live branch target
-  void cacheAssigns(std::list<Fortran::lower::pft::Evaluation> &evals) {
```

Comment:
I believe yes, pending actual complete implementation of bridge code.

Reply:
I'm not sure what is meant here by reachability.  The code as is generates unreachable blocks.  There is no reasonable alternative here, since GOTOs can appear almost anywhere.  Some downstream pass must remove unreachable blocks.  I don't think the code is generating gratuitously unnecessary unreachable blocks.

Reply:
(Having a pass to remove unreachable blocks is probably a good idea in any event, since it is likely that other optimization passes will generate dead code.)